core/modules/node/node.pages.inc

```diff
@@ -343,11 +341,22 @@ function node_form($form, &$form_state, Node $node) {
     '#submit' => array('node_form_submit'),
   );
   if (!empty($node->nid) && node_access('delete', $node)) {
+    $destination = array();
+    if (isset($_GET['destination'])) {
+      $destination = backdrop_get_destination();
+      unset($_GET['destination']);
+    }
```

Comment:
Everything looks good in this PR, with the exception that we shouldn't `unset($_GET['destination'])`. That has only removed in `node_form_delete_submit()` because forms have special handling in redirecting when `$_GET['destination']` is set.